earth_enterprise/src/common/khConfigFileParser.cpp

```diff
@@ -129,40 +129,44 @@ bool khConfigFileParser::isKeyPresent(const std::string& key)
 
 static khMutexBase xmlParmsLock = KH_MUTEX_BASE_INITIALIZER;
 
-void khConfigFileParser::parse(const std::string& fn)
+void khConfigFileParser::parse(std::stringstream& fileContents)
 {
-    std::ifstream file;
-    khLockGuard guard(xmlParmsLock);
-
-    contents.clear(); //clear out the old contents	
-    if (options.size() == 0) throw OptionsEmptyException(); // nothing to search on, exit
-
-    file.open(fn.c_str());
-    if (file.fail()) 
-    {
-        throw FileNotPresentException(fn);
-    }
-    
     std::string line;
-    while (getline(file,line))
+    contents.clear(); //clear out the old contents
+    while (getline(fileContents,line))
     {
         if (sanitize(line))
         {
             std::string key, value;
             split(line,key,value);
-            if (!isKeyPresent(key)) 
+            if (!isKeyPresent(key))
             {
-                file.close();
                 throw KeyNotPresentException(key);
             }
             if (value.size() == 0)
             {
-                file.close();
                 throw ValueNotPresentException(key);
             }
             contents.insert(std::pair<std::string,std::string>(key,value));
         }
     }
+}
+
+void khConfigFileParser::parse(const std::string& fn)
+{
+    std::ifstream file;
+    khLockGuard guard(xmlParmsLock);
+
+    if (options.size() == 0) throw OptionsEmptyException(); // nothing to search on, exit
+
+    file.open(fn.c_str());
+    if (file.fail()) 
+    {
+        throw FileNotPresentException(fn);
+    }
+    std::stringstream fileContents;
```

Comment:
Very minor: if you had the parse function called below take a `std::istream` as a parameter instead of a `std::stringstream` you could pass the `file` directly to the function instead of reading its contents into a stringstream first.